src/read_write/s2.rs

```diff
@@ -37,16 +40,21 @@ where
         let writers = LruCache::new(MAX_NUM_NODE_WRITERS);
         let already_opened_writers = HashSet::new();
         let stem = path.into();
+        let cell_stats = FnvHashMap::default();
```

Comment:
Nit: I would stay with the default `HashMap` in the future. Since Rust 1.36 it has a pretty fast implementation, see https://blog.rust-lang.org/2019/07/04/Rust-1.36.0.html#a-new-hashmapk,-v%3E-implementation.

Reply:
Yes, the HashMap algorithm itself is faster, but it still uses SipHash, FnvHashMap uses a different hash function, but the same HashMap algorithm optimized for small keys, so you get the benefits of both. If you prefer I can make this a HashMap though.